shared/types/Chat.ts

```diff
@@ -0,0 +1,11 @@
+import { ChatMessage } from ".";
+
+export interface Chat {
+  id: number;
+  messages: ChatMessage[];
+  /**
+   * Base contextual instruction to inform the LLM of its purpose/goal/role
+   */
+  contextInstruction?: string;
```

Comment:
`contextInstruction` is the base prompt ("You are a coding tutor" type stuff). Perhaps it should be named differently.

Yes, GPT does this prompt as a message, but PaLM does not. Not sure about other providers. When converting this `Chat` to the format expected by the LLMs, this `contextInstruction` is converted to a `ChatRole.System` (as opposed to `User` or `Assistant`) to send to GPT, but is included as the `context` field (separate from `examples` and `messages`) for the PaLM prompt. I don't know if not using PaLM's expected `context` argument and placing it in the messages directly meaningfully affects the LLM output though.

I think when we add the Personas feature, this base prompt will be wrapped up in the different person options. I envision the types being something like:
```typescript
export interface Chat {
  id: number;
  messages: ChatMessage[];
  persona: Persona
  title: string;
}

export interface Persona {
  title: string;
  basePrompt: string  // basePrompt or contextInstruction 
  // more probably
}
```

This way, the Persona/base prompt can be swapped out at any time. For instance, a user can change an existing conversation to be a Rubber Duck conversation instead of the regular assistant.